File: Lv 3/입국심사.py
def solution(n, times):
    answer = 9999999999
    r = max(times) * n
    l = 0
    
    while (l <= r):
        m = (l + r) // 2
        tot = 0
        for time in times:
            tot += (m // time)        

        if tot >= n:
            answer = m
            r = m - 1
        else:
            l = m + 1
    
    return answer

File: LV 2/입국심사.java
class Solution {
    public long solution(int n, int[] times) {
        long answer = 0;
        long left = 0;
        long right = 10000000000000L;
        while (left <= right) {
            long mid = (left + right) / 2;
            long sum = 0;
            for (int time : times) {
                sum += mid / time;
            }
            if (sum >= n) {
                right = mid - 1;
                answer = mid;
            } else {
                left = mid + 1;
            }
        }
        return answer;
    }
}
